### PR TITLE
Updates for the latest API of ouroboros-network

### PR DESCRIPTION
**Pull request by MarcFontaine**
* Update hashes in cabal.project and slack.yaml to latest versions
  of ouroboros-network and other dependencies. 
* Update then code to API changes in ouroboros-network.

# Issue Number

# Overview

- [ ] I have ...

# Comments

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -8,8 +8,93 @@ packages:
...
+    - typed-protocols
+    - typed-protocols-examples
+
+  - git: https://github.com/input-output-hk/cardano-node
+    commit: 3048967df2646cbe7519473600d044eeb2b0dfa8
+    subdirs: cardano-config
+
   - git: https://github.com/input-output-hk/cardano-db-sync
-    commit: 5407a064d4462712a5641295e93cf94d906bd897
+    commit: f37fbea644143b99060a6d068c679ba0be89cf4c
```
Comment (coot) [inline stack.yaml:97]: why so many changes in the `stack` file? Wasn't it upto date with `cabal.project` file?
If so it's worth checking if there is CI check enabled (if it is maybe it is not caputring the

File: submit-api/src/Cardano/TxSubmit/Node.hs [hunk 2]
```diff
@@ -226,12 +240,12 @@ nullChainSyncWithBlocksPtcl =
...
 txSubmissionClient
   :: TxSubmitVar -> TxSubmitMetrics
-  -> LocalTxSubmissionClient (GenTx ByronBlock) (ApplyTxErr ByronBlock) IO Void
+  -> LocalTxSubmissionClient (GenTx ByronBlock) (ApplyTxErr ByronBlock) IO ()
 txSubmissionClient tsv metrics =
     LocalTxSubmissionClient $
       readTxSubmit tsv >>= pure . loop
   where
-    loop :: GenTx ByronBlock -> LocalTxClientStIdle (GenTx ByronBlock) (ApplyTxErr ByronBlock) IO Void
+    loop :: GenTx ByronBlock -> LocalTxClientStIdle (GenTx ByronBlock) (ApplyTxErr ByronBlock) IO ()
```
Comment (coot) [inline submit-api/src/Cardano/TxSubmit/Node.hs:248]: Why it was necessary to switch from `Void` to `()`?

File: stack.yaml (hunk 1 above)
Comment (KtorZ) [reply stack.yaml:97]: The changes in the stack file are incorrect. We referenced the library in the snapshot already.

And no, as far as I know, there's no automated stack / cabal.project conversion going on in CI. Both have to be maintained separately

**Comment by KtorZ** [comment]
I'll have a deeper look at this in a few days / weeks. Thanks for updating this @MarcFontaine although this comes pretty low on our priority list. So, if the PR stays open, it's not because we have forget it, but simply because focus is elsewhere at the moment.

**Comment by MarcFontaine** [comment]
No problem.

File: submit-api/src/Cardano/TxSubmit/Node.hs (hunk 2 above)
Comment (MarcFontaine) [reply submit-api/src/Cardano/TxSubmit/Node.hs]: Right, I thinks it should be `Void`. I changed it

**Comment by rhyslbw** [comment]
Superseded by https://github.com/input-output-hk/cardano-rest/pull/42

**Pull request closed without merging**